hpc0.qmd

```diff
@@ -7,279 +7,1288 @@ format:
+:::
+
+
+## What is BASH? {.smaller}
+- An example of what it looks like:
 ![](assets/img/hpc0/bash.png)
 
-## What is a shell?
+::: {.notes}
+- `bash` is a particular type of shell.
+:::
+
 
+## What is a shell? 1/2 {.smaller}
 ![](assets/img/hpc0/shell.png)
 
-## MobaXTerm {background-color="#202020"}
+::: {.notes}
+- "Human users interface with the system via a shell."
+- "A shell communicates with the kernel."
+- "The kernel communicates with computer hardware and provides access the filesystem."
+- "There are a variety of Linux kernels and shells."
+:::
+
 
-Launch...
+## What is a shell? 2/2 {.smaller}
+- Shells have '`built-in`' commands:
+  - When called from a shell, `built-in` commands execute directly in the shell, without invoking (calling) other programs.
+- Shells call other programs by name, and these are found via the users '`PATH`' - a variable and ordered list of filesystem directories.
+- "On HPC resources at Leeds:
+   Additional Linux command utilities have been added and will be on your `PATH` for convenience.
+  - Additional software has been installed.
+    - Some of this is only available via a module system
+
+::: {.notes}
+- Read slide
+  - "Shells have built-in commands..."
+- "Details of the module system and how to use it are:"
+  - "Not covered in this course, but in HPC1."
+:::
 
-## Filesystem basics
 
+## Filesystem basics {.smaller}
 ![](assets/img/hpc0/filesystem.png)
 
-## Forward slash / and backslash \
+::: {.notes}
+- "The image shows a simplified filesystem."
+  - "We will look at what the actual file system looks like shortly..."
+- "A Linux 'Directory' is like a Windows 'Folder':"
+  - "A thing that can hold files and sub-folders/sub-directories."
+- "Directories are arranged in a heirarchy."
+  - "The top directory is the 'Root Directory'."
+- "Each user has a 'Home Directory':"
+  - "Where they login to."
+  - "Where their profile is stored."
+:::
+
 
+## Forward-slash / and back-slash \ {.smaller}
 ![](assets/img/hpc0/slash.png)
 
-## Filesystem basics: absolute path
+::: {.notes}
+- "The file path separator in Linux is the forward slash."
+- "The file path separator in Windows is the back slash."
+:::
+
 
+## Absolute path {.smaller}
 ![](assets/img/hpc0/filesystem-absolute.png)
 
-## Filesystem basics: relative path
+::: {.notes}
+- "The slide shows an absolute file path."
+- "The 'Root Directory' is referred to by a back slash."
+  - "All absolute file paths start from the 'Root Directory'..." 
+- Perhaps explain the file path further.
+- "Linux does not have drive letters like Windows."
+:::
 
+
+## Relative path {.smaller}
 ![](assets/img/hpc0/filesystem-relative.png)
 
-# Exercises {visibility="hidden"}
+::: {.notes}
+- "The slide shows a relative file path."
+- "'`..`' refers to one level up in the directory heirarchy."
+- Perhaps explain the file path further.
+:::
 
-## Exercise 0 {.smaller}
 
-### Logging in
+## The default prompt {.smaller}
```

Comment:
Single quote / code format tidy needed.  Notch this up as a general tidy/consistency fix needed throughout.